packages/neon_framework/lib/src/storage/sqlite_persistence.dart

```diff
@@ -0,0 +1,264 @@
+  ///
+  /// This must be called and completed before any sqlite persistence.
+  static Future<void> init() async {
+    if (database != null) {
+      return;
+    }
+
+    var path = 'preferences.db';
+    if (NeonPlatform.instance.canUsePaths) {
+      final appDir = await getApplicationSupportDirectory();
+      path = p.join(appDir.path, path);
+    }
+
+    database = await openDatabase(
+      path,
+      version: 1,
+      onCreate: onCreate,
+    );
+
+    await getAll();
+  }
+
+  @visibleForTesting
+  static Future<void> onCreate(Database db, int version) async {
+    await db.execute('''
+CREATE TABLE "preferences" (
+	"prefix"  TEXT NOT NULL,
+	"key"     TEXT NOT NULL,
+	"value"   TEXT NOT NULL,
+	PRIMARY KEY("prefix","key"),
+	UNIQUE("key","prefix")
+);
+''');
+  }
+
+  static Database get _requireDatabase {
+    if (database == null) {
+      throw StateError(
+        'Persistence has not been set up yet. Please make sure SQLitePersistence.init() has been called before and completed.',
+      );
+    }
+
+    return database!;
+  }
+
+  @visibleForTesting
+  static Future<void> getAll() async {
+    try {
+      final results = await _requireDatabase.rawQuery('''
+SELECT prefix, key, value
+FROM preferences
+''');
+
+      globalCache.clear();
+      for (final result in results) {
+        final prefix = result['prefix']! as String;
+        final key = result['key']! as String;
+        final value = result['value']! as String;
+
+        final cache = globalCache[prefix] ??= {};
+        cache[key] = _decode(value) as Object;
+      }
+    } on DatabaseException catch (error) {
+      _log.warning(
+        'Error fetching all values from the SQLite persistence.',
+        error,
+      );
+    }
+  }
+
+  @override
+  Future<bool> clear() async {
+    try {
+      await _requireDatabase.rawDelete(
+        '''
+DELETE FROM preferences
+WHERE prefix = ?
+''',
+        [prefix],
+      );
+      cache.clear();
+    } on DatabaseException catch (error) {
+      _log.warning(
+        'Error clearing the SQLite persistence.',
+        error,
+      );
+
+      return false;
+    }
+
+    return true;
+  }
+
+  @override
+  Future<void> reload() async {
+    try {
+      final fromSystem = <String, Object>{};
+
+      final results = await _requireDatabase.rawQuery(
+        '''
+SELECT key, value
+FROM preferences
+WHERE prefix = ?
+''',
+        [prefix],
+      );
+      for (final result in results) {
+        final key = result['key']! as String;
+        final value = result['value']! as String;
+
+        fromSystem[key] = _decode(value) as Object;
+      }
+
+      cache.clear();
+      cache.addAll(fromSystem);
+    } on DatabaseException catch (error) {
+      _log.warning(
+        'Error reloading the SQLite persistence.',
+        error,
+      );
+    }
+  }
+
+  @override
+  Future<bool> remove(String key) async {
+    cache.remove(key);
+
+    try {
+      await _requireDatabase.rawDelete(
```

Comment:
All write operations should only be executed on the cache in case the operation was successful on the DB.